miso-web/src/main/webapp/scripts/run_ajax.js

```diff
@@ -571,6 +558,64 @@ Run.ui = {
         }
       );
     }
+  },
+
+  addContainerByBarcode: function (runId) {
+    jQuery('#addRunNoteDialog')
+      .html("<form><p>Adding a container will lose any unsaved changes!</p>" +
+            "<fieldset class='dialog'>" +
+            "<label for='containerBarcode'></label>" +
+            "<input type='text' name='containerBarcode' id='containerBarcode' class='text ui-widget-content ui-corner-all' />" +
+            "</fieldset></form>");
+
+    jQuery('#addRunNoteDialog').dialog({
+      width: 400,
+      modal: true,
+      resizable: false,
+      buttons: {
+        "Add": function () {
+          Fluxion.doAjax(
+            'runControllerHelperService',
+            'addContainerByBarcode',
+            {
+              'runId': runId,
+              'barcode': jQuery('#containerBarcode').val(),
+              'url': ajaxurl
+            },
+            {
+              'doOnSuccess': function(json) {
+                if (json.err) {
```

Comment:
This can be true when `doOnSuccess` is called? I think there are other JSONUtil methods that will properly mark the response as error and invoke a different callback. Doesn't really matter, just an observation